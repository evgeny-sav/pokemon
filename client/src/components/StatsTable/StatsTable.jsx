import React from 'react';
import { Table } from 'react-bootstrap';
import capitalize from '../../utils/capitalize';
import { useSelector, shallowEqual } from 'react-redux';

// eslint-disable-next-line react/prop-types
const StatsTable = () => {
  const { id, height, weight, moves, stats, abilities } = useSelector(
    state => state.pokemon.data,
    shallowEqual,
  );

  return (
    <Table striped bordered hover>
      <thead>
        <tr>
          <th>Name</th>
          <th>Value</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>Id</td>
          <td>{id}</td>
        </tr>
        <tr>
          <td>Height</td>
          <td>{height}</td>
        </tr>
        <tr>
          <td>Weight</td>
          <td>{weight}</td>
        </tr>
        {/* eslint-disable-next-line react/prop-types */}
        {stats.map(stat => (
          <tr key={`${stat.stat.name}${stat.base_stat}`}>
            <td>{capitalize(stat.stat.name)}</td>
            <td>{stat.base_stat}</td>
          </tr>
        ))}
        {/* eslint-disable-next-line react/prop-types */}
        {abilities.map(ability => (
          <tr key={`${ability.ability.name}`}>
            <td>Ability:</td>
            <td>{capitalize(ability.ability.name)}</td>
          </tr>
        ))}
        {/* eslint-disable-next-line react/prop-types */}
        {moves.map(move => (
          <tr key={`${move.move.name}`}>
            <td>Move:</td>
            <td>{capitalize(move.move.name)}</td>
          </tr>
        ))}
      </tbody>
    </Table>
  );
};

export default StatsTable;
