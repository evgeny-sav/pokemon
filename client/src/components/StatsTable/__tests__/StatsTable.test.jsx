import React from 'react';
import { create } from 'react-test-renderer';
import StatsTable from '../StatsTable';

jest.mock('react-redux', () => ({
  useSelector: () => ({
    id: '',
    height: '',
    weight: '',
    moves: [],
    stats: [],
    abilities: [],
  }),
}));

describe('StatsTable', () => {
  it('should render', () => {
    const component = create(<StatsTable />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
