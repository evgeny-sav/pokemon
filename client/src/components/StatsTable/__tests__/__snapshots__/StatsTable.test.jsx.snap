// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsTable should render 1`] = `
<table
  className="table table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Value
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Id
      </td>
      <td>
        
      </td>
    </tr>
    <tr>
      <td>
        Height
      </td>
      <td>
        
      </td>
    </tr>
    <tr>
      <td>
        Weight
      </td>
      <td>
        
      </td>
    </tr>
  </tbody>
</table>
`;
